tt_metal/impl/program/program.hpp

```diff
@@ -50,6 +50,30 @@ struct KernelGroup {
     CoreType get_core_type() const;
 };
 
+struct transfer_info {
+    uint32_t size_in_bytes;
+    uint32_t dst;
+    uint32_t dst_noc_encoding;
+    uint32_t num_receivers;
+    bool last_transfer_in_group;
+    bool linked;
+};
+
+enum class PageTransferType { MULTICAST, UNICAST };
+
+struct ProgramDeviceMap {
```

Comment:
can you move this into the `.cpp`? It seems like users of `Program` aren't going to need to see this.

Reply:
It is a member of the program class. I could forward declare it if you'd prefer that.

Reply:
yes that would be great :), you'll need to have a pointer / reference to `ProgramDeviceMap` to enable forward decl

Reply:
FYI generally accepted name for this pattern is [PIMPL](https://en.cppreference.com/w/cpp/language/pimpl)

Reply:
I'm just wondering if this is the way to go... this will lead to an extra dereference each time we do anything with the program map. I think maybe cleanest would be for me to define the struct in a separate header so that people don't see it when looking in program.hpp.

Reply:
I just moved all dispatch program data structs/enums to a header, and pulling that into program.hpp. I think this is the cleanest solution.